unified-codes/apps/web/src/components/containers/error/ErrorPage.tsx

```diff
@@ -1,34 +1,63 @@
+
 import * as React from 'react';
+import { useParams } from 'react-router-dom';
+
+import { makeStyles, createStyles } from '@unified-codes/ui/styles';
+
+import ErrorNotFoundTop from './ErrorNotFoundTop';
+import ErrorNotFoundCenter from './ErrorNotFoundCenter';
+import ErrorNotFoundBottom from './ErrorNotFoundBottom';
+
+import ErrorLayout from '../../layout/ErrorLayout';
+
+import { Errors, IErrorRouteParams } from '../../../types';
+import { ITheme } from '../../../styles';
+
+const ERROR_CODES = {
+  [Errors.PageNotFound]: '404',
+}
 
-import { Grid } from '@unified-codes/ui/components';
+const useStyles = makeStyles((_: ITheme) => createStyles({
+  root: {
+      flexDirection: 'column',
+      alignContent: 'center',
+      padding: '20px 0 20px 0',
+  },
+  centerContainer: {
+      flexDirection: 'row',
+      justifyContent: 'center',
+  },
+  topContainer: {
+      textAlign: 'center'
+  },
+  bottomContainer: {
+    textAligh: 'center'
+  }
+}));
 
-const CatOnBike = require('../../../assets/cat-on-bike.png');
 
 export interface ErrorPageProps {
-  code: string;
+  onMount?: (params: IErrorRouteParams) => void;
+  onUnmount?: (params: IErrorRouteParams) => void;
 }
 
 export type ErrorPage = React.FunctionComponent<ErrorPageProps>;
 
-export const ErrorPage: ErrorPage = ({ code }) => {
-  switch (code) {
-    case '404':
-      return (
-        <Grid container style={{ flexDirection: 'column', alignContent: 'center' }}>
-          <Grid item style={{ textAlign: 'center' }}>
-            <h3>Nothing to see here...</h3>
-          </Grid>
-          <Grid item>
-            <img style={{ maxHeight: 'calc(100vh - 300px)', maxWidth: '100vw ' }} src={CatOnBike} />
-          </Grid>
-          <Grid item style={{ textAlign: 'center' }}>
-            <h3>...are you sure that you're in the right place?</h3>
-          </Grid>
-        </Grid>
-      );
-    default:
-      return <div>Oops! Something went wrong.</div>;
+export const ErrorPage: ErrorPage = ({ onMount = () => null, onUnmount = () => null }) => {
+  const classes = useStyles();
+  const params: IErrorRouteParams = useParams();
+
+  React.useEffect(() => {
+      onMount(params);
+      return () => onUnmount(params);
+  }, []);
+
+  const { code = ERROR_CODES[Errors.PageNotFound] } = params;
+
+  switch(code) {
+    case ERROR_CODES[Errors.PageNotFound]: return <ErrorLayout classes={classes} top={<ErrorNotFoundTop/>} center={<ErrorNotFoundCenter/>} bottom={<ErrorNotFoundBottom/>} />
```

Comment:
Would be good to add some more error pages, e.g. for authentication errors. Something for a separate issue?

Reply:
Yes, absolutely. I'd be keen to throw a `418` error - and perhaps there's already a good gif for that one?
Server error could be trickier, as `nginx` should raise / handle that. but a 403, surely? perhaps when we do admin - just to keep this PR moving along.